BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -827,7 +827,7 @@ public static float GetMaxThrust(Vessel v)
             engines.RemoveAll(e => !e.EngineIgnited || !e.isOperational);
             float thrust = engines.Sum(e => e.MaxThrustOutputVac(true));
 
-            List<ModuleRCSFX> RCS = VesselModuleRegistry.GetModules<ModuleRCSFX>(v);
+            List<ModuleRCSFX> RCS = VesselModuleRegistry.GetModules<ModuleRCSFX>(v).ToList();
```

Comment:
Taking a copy using `.ToList()` isn't necessary if you're not modifying the list or its entries. The issue with the engines was that you were removing the ones that weren't ignited or operational.
Looking at that bit again, it'd probably be better to write this function as
```csharp
public static float GetMaxThrust(Vessel v)
{
    float thrust = VesselModuleRegistry.GetModuleEngines(v).Where(e => e != null && e.EngineIgnited && e.isOperational).Sum(e => e.MaxThrustOutputVac(true));
    thrust += VesselModuleRegistry.GetModules<ModuleRCS>(v).Where(rcs => rcs != null && rcs.useThrottle).Sum(rcs => rcs.thrusterPower);
    return thrust;
}
```
to avoid allocating any extra memory.

Also, `VesselModuleRegistry.GetModules<ModuleRCSFX>(vessel)` on line 251 should be using `ModuleRCS` instead unless you're specifically wanting to do something with the FX.